src/core/pyproxy.c

```diff
@@ -223,6 +229,166 @@ _pyproxy_destroy(PyObject* ptrobj)
   EM_ASM({ delete Module.PyProxies[$0]; }, ptrobj);
 }
 
+/**
+ * Test if a PyObject is awaitable.
+ * Uses _PyCoro_GetAwaitableIter like in the implementation of the GET_AWAITABLE
+ * opcode (see ceval.c). Unfortunately this is not a public API (see issue
+ * https://bugs.python.org/issue24510) so it could be a source of instability.
```

Comment:
FWIW that function hasn't been modified in 6 years since it was added and it's still there in 3.10

Reply:
Thanks for checking! This isn't the only semi-public API we're using (included in the public headers but not documented, and starts with `_`), we also have:
`_PyType_Name` [added in version 3.7](https://bugs.python.org/issue31497)
`_PySet_Update`
`_PyList_Extend` 
`_Py_IDENTIFIER`
`_PyObject_CallMethodId`
`_PyObject_CallMethodIdObjArgs`
`_PyObject_GetAttrId`
`_PyObjectCall_NoArg` (stabilized to `PyObject_CallNoArgs` in v3.9)
`_Py_TPFLAGS_HAVE_VECTORCALL` (stabilized to `Py_TPFLAGS_HAVE_VECTORCALL` in v3.9)
`_PyErr_FormatFromCause`
`_PyGen_Send` ([going to be removed in v3.10](https://github.com/python/cpython/pull/22663))

Reply:
Probably should copy this info to somewhere more useful.

Reply:
Okay moved to #1251.